### PR TITLE
[#IOPID-1444] name inside mailvalidation

### PR DESCRIPTION
**Pull request by arcogabbo**
⚠️  This PR depends on #305, #306 ⚠️ 

**NOTE**: The `name` field needs to stay optional in openapi. Unfortunately, the `UpsertedProfileOrchestator` is called also in createProfile but we don't want to pass the `name` field there. Even if in the `UpsertedProfileOrchestrator` we don't call the `EmailValidationWithTemplateProcessOrchestrator` if the `profileOperation` equals to `CREATED`, the decode of the input would fail.

#### List of Changes

* [[#IOPID-1444] changed mailvalidation template script](https://github.com/pagopa/io-functions-app/commit/6a1016c93d9e67f1d7ba19a6a798f35ecc47866d)
* [[#IOPID-1444] changed openapi spec](https://github.com/pagopa/io-functions-app/commit/852cd04a65ea1c04a9d63863dc23917fb6f214e5)
* [[#IOPID-1444] added new script for mailvalidation](https://github.com/pagopa/io-functions-app/commit/8e08e60c41a133b6f109a37decdc8b24c5d6e5f7)
* [[#IOPID-1444] name inside mailvalidation](https://github.com/pagopa/io-functions-app/commit/419c6610b949d7858112bd5d7ad6ca4c0bcdb6f4)
* [[#IOPID-1444] update template version](https://github.com/pagopa/io-functions-app/pull/304/commits/23ad4ef7ce3e1503cf240099609078929022b188)
* [[#IOPID-1444] name format inside template](https://github.com/pagopa/io-functions-app/pull/304/commits/f37b88dd8bb7b743972356d760e6dc7c2f30053c)
* [[#IOPID-1444] name inside UpdateProfile](https://github.com/pagopa/io-functions-app/pull/304/commits/e436cfdbd0110426d9056e46f1776b37f0201d30)

#### Motivation and Context

add profile name inside validation mail

#### How Has This Been Tested?

integration test via io-mock

#### Screenshots (if appropriate):
image

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: UpsertedProfileOrchestrator/handler.ts [hunk 1]
```diff
@@ -49,6 +49,9 @@ export const OrchestratorInput = t.intersection([
     updatedAt: UTCISODateFromString
   }),
   t.partial({
+    // TODO: name field from partial to required after complete rollout of
+    // IOPID-1444 task
+    name: t.string,
```
Comment (gquadrati) [inline UpsertedProfileOrchestrator/handler.ts:54]: We should ensure that the new `name` property is added to all inputs of `UpsertedProfileOrchestrator`.

File: StartEmailValidationProcess/handler.ts [hunk 2]
```diff
@@ -159,7 +163,8 @@ export function StartEmailValidationProcess(
 
   const middlewaresWrap = withRequestMiddlewares(
     ContextMiddleware(),
-    FiscalCodeMiddleware
+    FiscalCodeMiddleware,
+    RequiredBodyPayloadMiddleware(EmailValidationProcessParams)
```
Comment (gquadrati) [inline StartEmailValidationProcess/handler.ts:167]: Have you tested the endpoint without including any request body? I haven't done so myself, but it's a precautionary measure, considering that the backend won't send any data until deployment.

File: _scripts/emailApplierTemplates/mailValidation [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const capitalize = (s: string): string =>
+  s.charAt(0).toUpperCase() + s.slice(1);
+
+const splitAndCap = (s: string): string =>
+  s
+    .toLowerCase()
+    .split(" ")
+    .map(e => capitalize(e))
+    .join(" ");
```
Comment (gquadrati) [inline _scripts/emailApplierTemplates/mailValidation:9]: Could we include some tests for this functionality?

**Review by gquadrati — CHANGES_REQUESTED** [review]

File: UpsertedProfileOrchestrator/handler.ts (hunk 1 above)
Comment (arcogabbo) [reply UpsertedProfileOrchestrator/handler.ts:54]: done in https://github.com/pagopa/io-functions-app/pull/304/commits/231f4842a7f9be6a3bf98b1577a31eacb377f31b

File: StartEmailValidationProcess/handler.ts (hunk 2 above)
Comment (arcogabbo) [reply StartEmailValidationProcess/handler.ts:167]: with #305, we will release the SDK so the backend can send the data before merging this PR

File: _scripts/emailApplierTemplates/mailValidation (hunk 3 above)
Comment (arcogabbo) [reply _scripts/emailApplierTemplates/mailValidation:9]: I added two new snapshots in https://github.com/pagopa/io-functions-app/pull/304/commits/3475c7bffc249d3aba733e237285f511494dedf1

**Review by gquadrati — APPROVED** [review]
LGTM 🚀

**Pull request merged**